src/course-outline/hooks.jsx

```diff
@@ -79,34 +91,54 @@ const useCourseOutline = ({ courseId }) => {
   };
 
   const handleOpenHighlightsModal = (section) => {
-    dispatch(setCurrentSection(section));
+    dispatch(setCurrentItem(section));
     openHighlightsModal();
   };
 
   const handleHighlightsFormSubmit = (highlights) => {
     const dataToSend = Object.values(highlights).filter(Boolean);
-    dispatch(updateCourseSectionHighlightsQuery(currentSection.id, dataToSend));
+    dispatch(updateCourseSectionHighlightsQuery(currentItem.id, dataToSend));
 
     closeHighlightsModal();
   };
 
-  const handlePublishSectionSubmit = () => {
-    dispatch(publishCourseSectionQuery(currentSection.id));
+  const handlePublishItemSubmit = () => {
+    dispatch(publishCourseItemQuery(currentItem.id, currentSection.id));
 
     closePublishModal();
   };
 
-  const handleEditSectionSubmit = (sectionId, displayName) => {
-    dispatch(editCourseSectionQuery(sectionId, displayName));
+  const handleEditSubmit = (itemId, sectionId, displayName) => {
+    dispatch(editCourseItemQuery(itemId, sectionId, displayName));
   };
 
-  const handleDeleteSectionSubmit = () => {
-    dispatch(deleteCourseSectionQuery(currentSection.id));
+  const handleDeleteItemSubmit = () => {
+    switch (currentItem.category) {
+    case COURSE_BLOCK_NAMES.chapter.id:
+      dispatch(deleteCourseSectionQuery(currentItem.id));
+      break;
+    case COURSE_BLOCK_NAMES.sequential.id:
+      dispatch(deleteCourseSubsectionQuery(currentItem.id, currentSection.id));
+      break;
+    case COURSE_BLOCK_NAMES.vertical.id:
+      dispatch(deleteCourseSubsectionQuery(
+        currentItem.id,
+        currentSubsection.id,
+        currentSection.id,
+      ));
```

Comment:
Is this in prevision of the Unit implementation? Should the deletion happen through `deleteCourseSubsectionQuery`?

Reply:
@CefBoud Nice catch! Missed to update it after copying.